incydr/cli/cmds/watchlists.py

```diff
@@ -0,0 +1,690 @@
+import click
+from click import Context
+from rich.panel import Panel
+from rich.progress import track
+
+from incydr._watchlists.models.responses import IncludedDepartment
+from incydr._watchlists.models.responses import IncludedDirectoryGroup
+from incydr._watchlists.models.responses import Watchlist
+from incydr._watchlists.models.responses import WatchlistUser
+from incydr.cli import console
+from incydr.cli import init_client
+from incydr.cli import log_file_option
+from incydr.cli import log_level_option
+from incydr.cli import render
+from incydr.cli.cmds.options.output_options import columns_option
+from incydr.cli.cmds.options.output_options import single_format_option
+from incydr.cli.cmds.options.output_options import SingleFormat
+from incydr.cli.cmds.options.output_options import table_format_option
+from incydr.cli.cmds.options.output_options import TableFormat
+from incydr.cli.cmds.options.utils import user_lookup_callback
+from incydr.cli.cmds.utils import user_lookup
+from incydr.cli.core import incompatible_with
+from incydr.cli.core import IncydrCommand
+from incydr.cli.core import IncydrGroup
+from incydr.enums.watchlists import WatchlistType
+from incydr.utils import model_as_card
+from incydr.utils import read_dict_from_csv
+
+
+# TODO: check for custom title?
+def get_watchlist_id_callback(ctx, param, value):
+    if not value:
+        return
+    try:
+        value = WatchlistType[value.upper()]  # if watchlist type
+        client = ctx.obj()
+        return client.watchlists.v1.get_id_by_name(value)
+    except KeyError:
+        return value
+
+
+@click.group(cls=IncydrGroup)
+@log_level_option
+@log_file_option
+@click.pass_context
+def watchlists(ctx, log_level, log_file):
```

Comment:
Each watchlist "membership category" (listed below as subcommand groups) has an `add`, `remove`, `list` and `show` command - with the exception of `members` which just has `list` and `show`.  

Right now I have this `watchlists` command group separated into those categories, where each category has all of those commands.  

ie: `watchlists included-users add <WATCHLIST-ID> <USER-IDS>

However I think another design for that we could potentially use is to only have the following commands

- list
- show
- add
- remove

And then have each of these accept flags that specify the group to add.  For example the following would add included and excluded users:

`watchlists add <WATCHLIST-ID>  --include-users <USER-IDS> --exclude-users <USER-IDS>`

This would probably require separating the csv functionality into a separate bulk command for each method as well, and would make all the different functionalities less visible/a little more complicated.  However, it would also cut down on the number of commands and allow users to do more in a single command.

Thoughts on these two designs?

Reply:
I like the idea of keeping most commands at the first sub-command level for visibility... what if we did just these:
```
watchlists list
watchlists show <watchlist>
watchlists list-members <watchlist>
watchlists add <watchlist> --users <user_ids|usernames> --departments <names> --directory-groups <names>
watchlists remove <watchlist> --users <user_ids|usernames> --departments <names> --directory-groups <names>
watchlists exclude-users <watchlist> <user_ids|usernames>
```

And `watchlists show` includes all of department/directory group data and 3 separate tables for members/included-users/excluded-users? CSV output for this command wouldn't really make sense, but we could have a nice `rich` layout of it all, and json for if someone _needed_ something programmatically from it.

For bulk add/remove/exclude, we could have the `watchlists add --users` option take a comma-separated string or `@filename` like we did in places in the code42cl, so it wouldn't necessarily need a separate command group.

Reply:
And I don't know how useful the individual "show" commands for members/departments/d-groups are, I think we can just leave those off for now.

Reply:
Yeah that seems like a good set of commands, where do we think 'remove excluded-users` would go in that layout?